aws_doc_sdk_examples_tools/lliam/service_layer/run_ailly.py

```diff
@@ -56,19 +64,27 @@ def resolve_requested_batches(batch_names: List[str]) -> List[Path]:
     return batch_paths
 
 
-def run_ailly_single_batch(batch: Path) -> None:
+def run_ailly_single_batch(batch: Path, packages: List[str] = []) -> None:
     """Run ailly and process files for a single batch."""
     batch_start_time = time.time()
     iam_updates_path = AILLY_DIR_PATH / f"updates_{batch.name}.json"
 
-    cmd = [
-        "ailly",
-        "--max-depth",
-        "10",
-        "--root",
-        str(AILLY_DIR_PATH),
-        batch.name,
-    ]
+    if packages:
+        paths = []
+        for package in packages:
+            package_files = [
+                f"{batch.name}/{p.name}" for p in batch.glob(f"*{package}*.md")
+            ]
+            paths.extend(package_files)
+
+        if not paths:
+            logger.warning(f"No matching files found for packages: {packages}")
+            return
```

Comment:
Isn't this OK? I'd let it be idempotent to missing packages.

Reply:
Ok!

Reply:
Is that worth an exit code > 0?

Reply:
Yeah probably. If we ever want to automate it a 0 would be unhelpful here.